### PR TITLE
Extra API to send goals from LUA/SC.

### PR DESCRIPTION
**Pull request by michalpelka**
It contains https://github.com/RobotecAI/o3de-humanworker-gem/pull/14 as well.

### CODE REVIEW

File: Code/Source/HumanWorker/NpcWaypointNavigatorComponent.cpp [hunk 1]
```diff
@@ -183,4 +190,21 @@ namespace ROS2::HumanWorker
         m_waypointEntities = waypointEntityIds;
     }
 
+    void NpcWaypointNavigatorComponent::GoToLocation(const AZ::EntityId& location)
+    {
+        if (m_waypointEntities.size()==1 && m_waypointEntities[0] == location)
```
Comment (arturkamieniecki) [inline Code/Source/HumanWorker/NpcWaypointNavigatorComponent.cpp]: Nit: formatting
```suggestion
        if (m_waypointEntities.size() == 1 && m_waypointEntities[0] == location)
```

File: Code/Source/HumanWorker/NpcWaypointNavigatorComponent.cpp [hunk 2]
```diff
@@ -183,4 +190,21 @@ namespace ROS2::HumanWorker
...
+        }
+        m_goalIndex = 0;
+        m_goalPath.clear();
+        m_state = NavigationState::Navigate;
+        m_waypointIndex = 0;
+        m_waypointEntities.clear();
+        m_waypointEntities.push_back(location);
+        m_restartOnTraversed = false;
+
+
```
Comment (arturkamieniecki) [inline Code/Source/HumanWorker/NpcWaypointNavigatorComponent.cpp]: Nit: Empty lines.

**Review by arturkamieniecki — APPROVED** [review]
Looks good!

File: Code/Source/HumanWorker/NpcWaypointNavigatorComponent.cpp [hunk 3]
```diff
@@ -183,4 +190,21 @@ namespace ROS2::HumanWorker
...
+            return;
+        }
+        m_goalIndex = 0;
+        m_goalPath.clear();
+        m_state = NavigationState::Navigate;
+        m_waypointIndex = 0;
+        m_waypointEntities.clear();
+        m_waypointEntities.push_back(location);
+        m_restartOnTraversed = false;
+
```
Comment (michalpelka) [inline Code/Source/HumanWorker/NpcWaypointNavigatorComponent.cpp]:
```suggestion
```

File: Code/Source/HumanWorker/NpcWaypointNavigatorComponent.cpp (hunk 2 above)
Comment (michalpelka) [inline Code/Source/HumanWorker/NpcWaypointNavigatorComponent.cpp]:
```suggestion
```

File: Code/Source/HumanWorker/NpcWaypointNavigatorComponent.cpp [hunk 4]
```diff
@@ -183,4 +190,19 @@ namespace ROS2::HumanWorker
...
+        }
+        m_goalIndex = 0;
+        m_goalPath.clear();
+        m_state = NavigationState::Navigate;
+        m_waypointIndex = 0;
+        m_waypointEntities.clear();
+        m_waypointEntities.push_back(location);
+        m_restartOnTraversed = false;
+    }
+
```
Comment (michalpelka) [inline Code/Source/HumanWorker/NpcWaypointNavigatorComponent.cpp]:
```suggestion
```

**Pull request merged**